pkg/prometheus/server/statefulset.go

```diff
@@ -730,6 +730,12 @@ func createThanosContainer(
 		if thanos.ReadyTimeout != "" {
 			thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "prometheus.ready_timeout", Value: string(thanos.ReadyTimeout)})
 		}
+		if thanos.GetConfigTimeout != "" {
```

Comment:
We need to check for `thanos.Version` as the flag requires v0.28.0 at least.

Reply:
I've added the hashicorp version checker. If you have something specific in mind, I can also implement that instead.

Reply:
We already use `github.com/blang/semver/v4` for this. See for https://github.com/prometheus-operator/prometheus-operator/blob/dd19613873f75c402d64ec179433a3531a0cbe4a/pkg/prometheus/promcfg.go#L68-L72 an example